lib/attachment_processors/image.rb

```diff
@@ -0,0 +1,40 @@
+require 'tempfile'
+require 'securerandom'
+require 'rmagick'
+require 's3_uploader'
+
+module AttachmentProcessor
+  class Image
+    attr_reader :attachment, :ext, :original_img, :resize_img
+
+    def initialize(attachment)
+      @attachment = attachment
+      @ext = ".#{attachment.extension}"
+      @original_img = Tempfile.new([SecureRandom.hex(18), ext])
+      @resize_img = Tempfile.new([SecureRandom.hex(18), ext])
+
+      open(attachment.url) do |file|
+        original_img.write(file.read.force_encoding("utf-8"))
+      end
+    end
+
+    def resize_and_upload(width: 600, height: 400, tag: :thumb)
+      # resize it
+      img = Magick::Image.read(original_img.path).first
```

Comment:
I think rmagick/imagemagick can open images directly from an http url, we may not need to download the image before opening it into rmagick